src/history_be/main.mo

```diff
@@ -235,21 +213,49 @@ actor class HistoryTracker() = self {
     var callsToSpawn = Int.abs(Int.max(0, canisters_num_to_sync - open_calls));
     var spawnedCalls = 0;
 
+    let calls = Buffer.Buffer<Concurrent.Item>(callsToSpawn);
+
+    func callItem(canisterIdx : Nat, register_fail_cb : () -> ()) : () {
```

Comment:
Do we need this function definition inside `trigger_sync()` or can it be defined before, outside?

Reply:
We can do either, because this function is used only here. Do you think that defining it outside gives some benefits?

Reply:
Only readability because it makes the containing function shorter.

Reply:
Done